src/redux/books/books.js

```diff
@@ -1,49 +1,40 @@
-import axios from 'axios';
-
-const ADD_BOOK = 'react-bookstore/books/ADD_BOOK';
-const DELETE_BOOK = 'react-bookstore/books/DELETE_BOOK';
-const BOOK_FAILURE = 'react-bookstore/books/BOOK_FAILURE';
-
-const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps';
-const identifier = 'pSmXMG8pCv1BQZweEpXG';
-
-const books = [];
-
-const BookReducer = (state = books, action) => {
-  switch (action.type) {
-    case ADD_BOOK: {
-      const books = Object.entries(action.payload);
-      return books.map((book) => ({
-        id: book[0],
-        ...book[1][0],
-      }));
-    }
-    case DELETE_BOOK:
-      return [...state.filter((book) => (book.id !== action.payload))];
-    default:
-      return state;
-  }
-};
-
-export const addBook = (book) => ({ type: ADD_BOOK, payload: book });
-
-export const getBooks = () => async (dispatch) => {
-  await axios.get(`${url}/${identifier}/books`).then(
-    (response) => dispatch(addBook(response.data)),
-    (err) => dispatch({ type: BOOK_FAILURE, err }),
-  );
-};
-
-export const deleteBook = (id) => async (dispatch) => {
-  await axios
-    .delete(`${url}/${identifier}/books/${id}`)
-    .then(() => dispatch(getBooks())); return { type: DELETE_BOOK, payload: id };
-};
-
-export const postBook = (book) => async (dispatch) => {
-  await axios
-    .post(`${url}/${identifier}/books`, book)
-    .then(() => dispatch(getBooks()));
-};
-
-export default BookReducer;
+import { createSlice } from '@reduxjs/toolkit';
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState: [
+    {
+      item_id: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
+  reducers: {
+    addBook: (state, action) => {
+      const newBook = {
+        id: Date.now(),
+        title: action.payload.title,
+        author: action.payload.author,
+      };
+      state.push(newBook);
+    },
+    removeBook: (state, action) => state.filter((book) => book.key !== action.payload.id),
+  },
```

Comment:
- [ ] [Optional] When I try to remove a single book then every book disappears please see this issue and fix it. you can try like this

` import React from "react"
import { useDispatch } from "react-redux"
import { removeBook } from "path/to/booksSlice"

function Book({ id, title, author }) {
  const dispatch = useDispatch()

  return (
    <div>
      {/* Rest of Book JSX */}
      <Button onClick={() => dispatch(removeBook(id))}>Remove</Button>
    </div>
  )
}

export default Book`